main.nf

```diff
@@ -0,0 +1,932 @@
+        saveAs: {filename -> filename.indexOf(".zip") > 0 ? "zips/$filename" : "$filename"}
+
+    input:
+    set val(name), file(reads) from read_files_fastqc
+
+    output:
+    file '*_fastqc.{zip,html}' into fastqc_results
+    file '.command.out' into fastqc_stdout
+
+    script:
+    """
+    fastqc -q $reads
+    """
+}
+
+
+/*
+ * STEP 1 - trim with trim galore
+ */
+
+if(params.notrim){
+    trimmed_reads = read_files_trimming
+    trimgalore_results = []
+} else {
+    process trim_galore {
+        tag "$name"
+        publishDir "${params.outdir}/trim_galore", mode: 'copy'
+
+        input:
+        set val(name), file(reads) from read_files_trimming
+
+        output:
+        set val(name), file('*fq.gz') into trimmed_reads
+        file '*trimming_report.txt' into trimgalore_results, trimgalore_logs
+
+        script:
+        single = reads instanceof Path
+        c_r1 = params.clip_r1 > 0 ? "--clip_r1 ${params.clip_r1}" : ''
+        c_r2 = params.clip_r2 > 0 ? "--clip_r2 ${params.clip_r2}" : ''
+        tpc_r1 = params.three_prime_clip_r1 > 0 ? "--three_prime_clip_r1 ${params.three_prime_clip_r1}" : ''
+        tpc_r2 = params.three_prime_clip_r2 > 0 ? "--three_prime_clip_r2 ${params.three_prime_clip_r2}" : ''
+        if (params.singleEnd) {
+            """
+            trim_galore --gzip $c_r1 $tpc_r1 $reads
+            """
+        } else {
+            """
+            trim_galore --paired --gzip $c_r1 $c_r2 $tpc_r1 $tpc_r2 $reads
+            """
+        }
+    }
+}
+
+
+/*
+ * STEP 2 - Map with BWA Mem
+ */
+
+process bwamem {
+    tag "$name"
+
+    input:
+    set val(name), file(reads) from trimmed_reads
+    file(bwa_index) from bwa_index
+
+    output:
+    set val(name), file("${name}_bwa.sam") into raw_aln_sam
+    file '.command.log' into bwa_stdout
+
+
+    script:
+    rg="\'@RG\\tID:${params.run_id}\\tSM:${params.run_id}\\tPL:illumina\'"
+    if(params.singleEnd){
+    """
+         bwa mem \\
+        -R $rg \\
+        -t ${task.cpus} \\
+        -k 2 \\
+        $params.gfasta \\
+        $reads \\
+            > ${name}_bwa.sam
+
+    # Print version number to standard out
+    echo "BWA Version:"\$(bwa 2>&1)
+    """
+    } else {
+    """
+        bwa mem \\
+        -R $rg \\
+        -t ${task.cpus} \\
+        -k 2 \\
+        $params.gfasta \\
+        $reads\\
+            > ${name}_bwa.sam
+
+    # Print version number to standard out
+    echo "BWA Version:"\$(bwa 2>&1)
+    """
+
+}
+}
+
+/*
+*  STEP 3 - Convert to BAM, sort BAM  
+*/ 
+
+process sortSam {
+    tag "${name}"
+    publishDir "${params.outdir}/BWAmem", mode: 'copy',
+        saveAs: {filename -> params.saveAlignedIntermediates ? "$filename" : null }
+    
+    input:
+    set val(name), file(raw_sam) from raw_aln_sam
+    
+
+    output: 
+    set val(name), file("${raw_sam}.sorted.bam") into samples_sorted_bam
+
+    script:
+    def avail_mem = task.memory == null ? '' : "-m ${Math.floor(task.memory.toBytes() / task.cpus)}"
```

Comment:
It may be worth wrapping this in a `try` /  `except` block. `task.memory` and `task.cpus` may not always be present - for example on the hebbe cluster, jobs are rejected if you try to request a specific amount of memory.

Reply:
I added such a statement and the catch statement should produce an output that is then logged by NXF - should be good.